src/scripts/main.js

```diff
@@ -1,3 +1,210 @@
+}
+
+function getCellValue() {
+  return Math.random() < 0.9 ? 2 : 4;
+}
+
+function newCell() {
+  if (clearCells.length === undefined) {
+    return;
+  }
+
+  const randomCell = getRandomCell();
+  const randomValue = getCellValue();
+
+  randomCell.className = `field-cell field-cell--${randomValue}`;
+  randomCell.innerText = randomValue;
+
+  clearCells.splice(clearCells.indexOf(randomCell), 1);
+}
+
+let started = false;
+
+function startGame() {
+  startMessage.style.display = 'none';
+  button.className = 'button restart';
+  button.innerText = 'Restart';
+
+  newCell();
+  newCell();
+  started = true;
+}
+
+function restartGame() {
+  loseMessage.classList.add('hidden');
+  winMessage.classList.add('hidden');
+  clearCells = [...cells];
+  score.innerText = 0;
+
+  cells.forEach(cell => {
+    cell.innerText = '';
+    cell.className = 'field-cell';
+  });
+
+  newCell();
+  newCell();
+}
+
+button.addEventListener('click', () => {
+  if (!started) {
+    startGame();
+  } else {
+    restartGame();
+  }
+});
+
+function canBeMarged() {
+  const wholeField = [...cellsInRow, ...columns];
+
+  for (const line of wholeField) {
+    for (let i = 0; i < 3; i++) {
+      if (line[i].innerText === line[i + 1].innerText) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+let flipped = false;
+
+function flipLine(line) {
+  for (let i = 3; i > 0; i--) {
+    const canBeMooved = clearCells.includes(line[i])
+      && !clearCells.includes(line[i - 1]);
+    const mergeAllowed = line[i].innerText === line[i - 1].innerText
+      && line[i].innerText.length
+      && !line[i - 1].dataset.blocked;
+
+    if ((mergeAllowed || canBeMooved) && !flipped) {
+      flipped = true;
+    }
+
+    if (mergeAllowed) {
+      mergeCells(line[i], line[i - 1]);
+      flipLine(line);
+    }
+
+    if (canBeMooved) {
+      flipElements(line[i], line[i - 1]);
+      flipLine(line);
+    }
+  }
+}
+
+function mergeCells(curr, prev) {
+  const value = curr.innerText * 2;
+
+  curr.innerText = value;
+  curr.className = `field-cell field-cell--${value}`;
+  score.innerText = +score.innerText + value;
+  deleteElement(prev);
+
+  curr.dataset.blocked = true;
+  prev.dataset.blocked = true;
+
+  if (value === 2048) {
+    winMessage.classList.remove('hidden');
+    winSound.currentTime = 0;
+    winSound.play();
+  }
+}
+
+function flipElements(curr, prev) {
+  curr.innerText = prev.innerText;
+  curr.className = prev.className;
+  clearCells = clearCells.filter(cell => cell !== curr);
+  deleteElement(prev);
+}
+
+function deleteElement(element) {
+  element.innerText = '';
+  element.className = 'field-cell';
+  clearCells.push(element);
+}
+
+function getDirection(direction) {
+  if (!clearCells.length && !canBeMarged()) {
+    loseMessage.classList.remove('hidden');
+    looseSound.currentTime = 0;
+    looseSound.play();
+  }
+
+  switch (direction) {
+    case 'ArrowUp':
+      for (const column of columns) {
```

Comment:
Avoid using a `for...of` loop to iterate over an array. Instead, use `forEach` or `map`.